examples/qa-automation/qa_automation.py

```diff
@@ -0,0 +1,240 @@
+- Do not generate asserts for 'Given', 'When', or 'And' steps; only generate asserts for 'Then' steps.
+- Always use provided selenium code that was already executed to find valid selectors for the final pytestfile. 
+- Never output antyhing other than markdown python code.
+"""
+
+EXAMPLES = """
+import pytest
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from pytest_bdd import scenarios, given, when, then
+
+# Constants
+BASE_URL = 'https://form.jotform.com/241472287797370'
+
+# Scenarios
+scenarios('test_form_submission.feature')
+
+# Fixtures
+@pytest.fixture
+def browser():
+    driver = webdriver.Chrome()
+    driver.implicitly_wait(10)
+    driver.get(BASE_URL)
+    yield driver
+    driver.quit()
+
+# Steps
+@given('I am on the job application page')
+def i_am_on_the_job_application_page(browser):
+    pass
+
+@when('I enter "John" in the "First Name" field')
+def i_enter_first_name(browser):
+    first_name_field = browser.find_element(By.XPATH, "/html/body/form/div[1]/ul/li[2]/div/div/span[1]/input")
+    first_name_field.send_keys("John")
+
+@when('I enter "Doe" in the "Last Name" field')
+def i_enter_last_name(browser):
+    last_name_field = browser.find_element(By.XPATH, "/html/body/form/div[1]/ul/li[2]/div/div/span[2]/input")
+    last_name_field.send_keys("Doe")
+
+@when('I enter "john.doe@example.com" in the "Email Address" field')
+def i_enter_email_address(browser):
+    email_field = browser.find_element(By.XPATH, "/html/body/form/div[1]/ul/li[3]/div/span/input")
+    email_field.send_keys("john.doe@example.com")
+
+@when('I enter "(123) 456-7890" in the "Phone Number" field')
+def i_enter_phone_number(browser):
+    phone_number_field = browser.find_element(By.XPATH, "/html/body/form/div[1]/ul/li[4]/div/span/input")
+    phone_number_field.send_keys("(123) 456-7890")
+
+@when('I leave the "Cover Letter" field empty')
+def i_leave_cover_letter_empty():
+    # No action needed as the field should remain empty
+    pass
+
+@when('I click the "Apply" button')
+def i_click_apply_button(browser):
+    apply_button = WebDriverWait(browser, 10).until(
+        EC.element_to_be_clickable((By.XPATH, "/html/body/form/div[1]/ul/li[6]/div/div/button"))
+    )
+    browser.execute_script("arguments[0].scrollIntoView(true);", apply_button)
+    apply_button.click()
+
+@then('I should see an error message for the "Cover Letter" field')
+def i_should_see_error_message(browser):
+    try:
+        error_message = browser.find_element(By.XPATH, "/html/body/form/div[1]/ul/li[5]/div/div/span")
+        assert error_message.is_displayed()
+    except Exception as e:
+        pytest.fail(f"Error message not displayed: {e}")
+"""
+
+if __name__ == "__main__":
+    os.environ["OPENAI_API_KEY"] = fetch_secret("OPENAI_API_KEY")
```

Comment:
`NameError: name 'fetch_secret' is not defined` when I run this locally. Locally, it should automatically be able to take OPENAI_API_KEY from your env so you only want to get the secret IF running in Colab, which you can do with:

`import os

# Check if running in Google Colab
try:
    from google.colab import userdata
    IN_COLAB = True
except ImportError:
    IN_COLAB = False

if IN_COLAB:
    os.environ["OPENAI_API_KEY"] = userdata.get('OPENAI_API_KEY')`